src/globalvars.cpp

```diff
@@ -258,19 +308,99 @@ void GlobalVars::verify_maps_found(
     const std::unordered_map<std::string, struct bpf_map *>
         &section_name_to_global_vars_map)
 {
-  for (const auto &[name, config] : global_var_map_) {
+  for (const auto &[name, config] : added_global_vars_) {
     if (!section_name_to_global_vars_map.contains(config.section)) {
       LOG(BUG) << "No map found in " << config.section
                << " which is needed to set global variable " << name;
     }
   }
 }
 
+Result<GlobalVarVals> GlobalVars::get_named_param_vals(
+    std::unordered_map<std::string, std::string> named_params)
+{
+  GlobalVarVals named_param_vals;
+
+  std::string expected_args_str;
+  for (const auto &[name, default_val] : named_param_defaults_) {
+    auto &config = added_global_vars_.at(name);
+    auto it = named_params.find(name);
+    std::string val = default_val;
+
+    if (it != named_params.end()) {
+      val = it->second;
+      named_params.erase(name);
+    }
+
+    if (config.type == GlobalVarType::integer) {
+      try {
+        named_param_vals[name] = std::stoll(val);
+      } catch (const std::invalid_argument &) {
+        return make_error<NamedParamError>(
+            "Command line option --" + name +
+            " is expecting an integer. Got: " + val);
+      } catch (const std::out_of_range &) {
+        return make_error<NamedParamError>("Value for command line option --" +
+                                           name +
+                                           " is out of range. Got: " + val);
+      }
+    } else if (config.type == GlobalVarType::boolean) {
+      if (val.empty()) {
+        named_param_vals[name] = 1;
+      } else if (util::is_str_bool_truthy(val)) {
+        named_param_vals[name] = 1;
+      } else if (util::is_str_bool_falsy(val)) {
+        named_param_vals[name] = 0;
+      } else {
+        return make_error<NamedParamError>(
+            "Command line option --" + name +
+            " is expecting a boolean (e.g. 'true'). Got: " + val);
+      }
+    } else if (config.type == GlobalVarType::string) {
+      if (val.empty()) {
+        return make_error<NamedParamError>(
+            "Command line option --" + name +
```

Comment:
I know there's been some discussion on this, but errors are not paragraphs. Don't capitalize, and don't use complete sentences that end with periods. It is a fragment that describes the error, if you want richer structure, you need to compose using the type itself, e.g. `NamedParamError` shouldn't just be a string, it can have a parameter name **and** and error message, and those can be composed in it's `log` function

Reply:
Made some updates here, let me know if it's what you had in mind.